api/app/Console/Commands/MigratePoolStream.php

```diff
@@ -0,0 +1,111 @@
+ *
+ *  TODO: Remove in #12142
+ */
+class MigratePoolStream extends Command
+{
+    /**
+     * The name and signature of the console command.
+     *
+     * @var string
+     */
+    protected $signature = 'app:migrate-pool-stream';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Migrates the pool stream enum to work stream model';
+
+    /**
+     * Execute the console command.
+     */
+    public function handle()
+    {
+
+        $workStreams = WorkStream::all();
+
+        $this->info('Updating job poster template work streams...');
+        $jobPosterTemplatesUpdated = 0;
+        JobPosterTemplate::whereNotNull('stream')->chunkById(200, function (Collection $jobPosterTemplates) use ($workStreams, &$jobPosterTemplatesUpdated) {
+            foreach ($jobPosterTemplates as $jobPosterTemplate) {
+                $stream = $workStreams->where('key', $jobPosterTemplate->stream)->first();
+                if ($stream) {
+                    $jobPosterTemplate->work_stream_id = $stream->id;
+                    $jobPosterTemplate->save();
+                    $jobPosterTemplatesUpdated++;
+                } else {
+                    $this->error(sprintf('Work stream (%s) not found for job poster template (%s)',
+                        $jobPosterTemplate->stream,
+                        $jobPosterTemplate->id));
+                }
+            }
+        });
+        $this->info("Updated $jobPosterTemplatesUpdated job poster templates");
+
+        $this->newLine();
+        $this->info('Updating pool work streams...');
+        $poolsUpdated = 0;
+        Pool::whereNotNull('stream')->chunkById(200, function (Collection $pools) use ($workStreams, &$poolsUpdated) {
+            foreach ($pools as $pool) {
+                $stream = $workStreams->where('key', $pool->stream)->first();
+                if ($stream) {
+                    $pool->work_stream_id = $stream->id;
+                    $pool->save();
+                    $poolsUpdated++;
+                } else {
+                    $this->error(sprintf('Work stream (%s) not found for pool (%s)',
+                        $pool->stream,
+                        $pool->id));
+                }
+            }
+        });
+        $this->info("Updated $poolsUpdated pools");
+
+        $applicantFiltersUpdated = 0;
+        $this->newLine();
+        $this->info('Updated applicant filter work streams...');
+        ApplicantFilter::whereNotNull('qualified_streams')->chunkById(200, function (Collection $applicantFilters) use ($workStreams, &$applicantFiltersUpdated) {
+            foreach ($applicantFilters as $applicantFilter) {
+                $streams = $workStreams->whereIn('key', $applicantFilter->qualified_streams)->pluck('id');
+                if ($streams) {
+                    $applicantFilter->workStreams()->sync($streams);
+                    $applicantFiltersUpdated++;
+                } else {
```

Comment:
This conditional block doesn't seem to work as expected, `$streams` always gets treated as true so the updated count always matches the number of filters stored even if those filters aren't connected to any streams. The variable resolves to an empty array if nothing was found

![image](https://github.com/user-attachments/assets/708cae00-0021-41a4-8937-c88f8511f701)

Subsequently, a non-existent stream is not being flagged for applicant filters

The oddity in the conditional

![image](https://github.com/user-attachments/assets/e1e54d2f-c106-4cfb-89e3-5b46222c524a)

![image](https://github.com/user-attachments/assets/0582d0ca-da19-4722-aeb6-f1449146a296)